eZ/Publish/Core/Persistence/Legacy/Content/ObjectState/Gateway/DoctrineDatabase.php

```diff
@@ -1,760 +1,624 @@
-        )->set(
-            $this->dbHandler->quoteColumn('language_mask'),
-            $query->bindValue(
-                $this->maskGenerator->generateLanguageMaskFromLanguageCodes($objectState->languageCodes, true),
-                null,
-                ParameterType::INTEGER
-            )
-        )->set(
-            $this->dbHandler->quoteColumn('priority'),
-            $query->bindValue($objectState->priority, null, \PDO::PARAM_INT)
-        );
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->insert(self::OBJECT_STATE_TABLE)
+            ->values(
+                [
+                    'group_id' => $query->createPositionalParameter(
+                        $objectState->groupId,
+                        ParameterType::INTEGER
+                    ),
+                    'default_language_id' => $query->createPositionalParameter(
+                        $this->maskGenerator->generateLanguageIndicator(
+                            $objectState->defaultLanguage,
+                            false
+                        ),
+                        ParameterType::INTEGER
+                    ),
+                    'identifier' => $query->createPositionalParameter(
+                        $objectState->identifier,
+                        ParameterType::STRING
+                    ),
+                    'language_mask' => $query->createPositionalParameter(
+                        $this->maskGenerator->generateLanguageMaskFromLanguageCodes(
+                            $objectState->languageCodes,
+                            true
+                        ),
+                        ParameterType::INTEGER
+                    ),
+                    'priority' => $query->createPositionalParameter(
+                        $objectState->priority,
+                        ParameterType::INTEGER
+                    ),
+                ]
+            );
 
-        $query->prepare()->execute();
+        $query->execute();
 
-        $objectState->id = (int)$this->dbHandler->lastInsertId(
-            $this->dbHandler->getSequenceName('ezcobj_state', 'id')
-        );
+        $objectState->id = (int)$this->connection->lastInsertId(self::OBJECT_STATE_TABLE_SEQ);
 
         $this->insertObjectStateTranslations($objectState);
 
         // If this is a first state in group, assign it to all content objects
         if ($maxPriority === null) {
-            // @todo Hm... How do we perform this with query object?
-            $this->dbHandler->prepare(
-                "INSERT INTO ezcobj_state_link (contentobject_id, contentobject_state_id)
-                SELECT id, {$objectState->id} FROM ezcontentobject"
-            )->execute();
+            $this->connection->executeUpdate(
+                'INSERT INTO ezcobj_state_link (contentobject_id, contentobject_state_id) ' .
+                "SELECT id, {$objectState->id} FROM ezcontentobject"
+            );
```

Comment:
> Parameters should be passed to query using binding instead of the string interpolation. Could you please change it?

Thx, fixed via 412daeb.

Reply:
> > Parameters should be passed to query using binding instead of the string interpolation. Could you please change it?
> 
> Thx, fixed via [412daeb](https://github.com/ezsystems/ezpublish-kernel/commit/412daeb2b7ed9447b94e002cb001bb2eb34d80a2).

Unfortunately MySQL driver does not support binding parameters for SELECT part, so I had to revert it. This was my concern when writing this initially, but I couldn't remember the exact reason. When using _normal_[1] drivers for _normal_[1] databases, binding works as it should.

[1] PostgreSQL